dataflux_pytorch/benchmark/checkpointing/simulated/multi_node_benchmark.py

```diff
@@ -49,56 +44,64 @@ def init_processes() -> int:
+            state_dict[f'dummy_tensor_{i}'] = torch.randn(layer_size, 1000)
+    dist.barrier()
+
+    save_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'save',
+                                                      rank, world_size,
+                                                      padding_size, layer_size)
+
+    load_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'load',
+                                                      rank, world_size,
+                                                      padding_size, layer_size)
 
     if rank == 0:
-        print("######################")
-        print(
-            f"Time taken to save checkpoint: {statistics.mean(save_checkpoint_times):.4f} seconds")
-        print(
-            f"Time taken to load checkpoint: {statistics.mean(load_checkpoint_times):.4f} seconds")
-        total_distributed_size_bytes = sum(get_tensor_size_bytes(
-            tensor) for tensor in distributed_state_dict.values())
-        print(
-            f"Size of distributed tensors (rank {rank}): {format_size(total_distributed_size_bytes)}")
-        print(
-            f"Total size of all tensors (rank {rank}): {format_size(total_distributed_size_bytes * world_size)}")
+        print(f"Time taken to save checkpoint:\
+                {statistics.mean(save_checkpoint_times):.4f} seconds")
+        print(f"Time taken to load checkpoint:\
+                 {statistics.mean(load_checkpoint_times):.4f} seconds")
+
+        tensor_size_per_instance = 1000 * layer_size * state_dict[
+            f'dummy_tensor_0'].element_size()
+        total_tensor_count = padding_size // world_size  # Total tensors per rank
+        total_size_bytes = total_tensor_count * tensor_size_per_instance * world_size
+        print(f"Size of distributed tensors (rank {rank}):\
+                 {format_size(total_tensor_count * tensor_size_per_instance)}")
+        print(f"Total size of all tensors:\
+                 {format_size(total_size_bytes)}")
         print("######################")
 
     cleanup()
 
 
 if __name__ == "__main__":
     world_size = int(os.getenv("WORLD_SIZE"))
-    model_size = int(os.getenv("NUM_LAYERS"))
+    layer_size = int(os.getenv("LAYER_SIZE"))
     project = os.getenv("PROJECT")
-    path = os.getenv("CKPT_DIR_PATH")
-    sample_size = int(os.getenv("SAMPLE_COUNT", 3))
+    ckpt_dir_path = os.getenv("CKPT_DIR_PATH")
+    sample_count = int(os.getenv("SAMPLE_COUNT", 3))
     padding_size = int(os.getenv("PADDING_SIZE", 4000))
-    main(world_size, model_size,
-         project, path, padding_size, sample_size)
+    main(world_size, layer_size, project, ckpt_dir_path, padding_size,
```

Comment:
main() shouldn't really be taking arguments. We should move these environment variable grabs into a setup function and call it from main.

Reply:
We already have a different version under demo for kokoro integration https://github.com/GoogleCloudPlatform/dataflux-pytorch/tree/main/demo/lightning/checkpoint/simulated , so I guess this can remain as it is ? 

Offcourse when someone takes on b/2F367371602, they can decide on long term strategy on how to combine code between benchmarks and demo.

Reply:
Not sure if there is any standard or conventional style in Python for this, but my sense is that main usually should not be grabbing environment variables (i.e. so that it could be called as library code from somewhere else if you wanted to) and this is a good place to do it.

Reply:
If we want to keep the env var args here thats ok, but we should still not pass individual args in main. Instead we can pass an argv single param containing a sequence like here: https://engdoc.corp.google.com/eng/doc/devguide/py/style/index.md?cl=head

Reply:
I can do something like this https://github.com/GoogleCloudPlatform/dataflux-pytorch/blob/main/demo/lightning/checkpoint/simulated/multiprocessing_train.py#L356 

main doesnt take any args, but it parses env var and then calls run_benchmark. Sounds good ? @jdnurme @MattIrv

Reply:
Yeah that looks fine to me. Longterm we can also create some sort of config object for this that we can share across all demo code, but that's definitely out of scope for this. We will eventually need to sort out the mix of command line vs env var stuff we have.

Reply:
Done